src/rust/dbv2/migrations/2023-11-05-124149_recognized_market_endpoint/up.sql

```diff
@@ -0,0 +1,58 @@
+-- Your SQL goes here
+CREATE VIEW api.recognized_markets AS
+    SELECT
+    DISTINCT ON (
+        base_account_address,
+        base_module_name,
+        base_struct_name,
+        base_name_generic,
+        quote_account_address,
+        quote_module_name,
+        quote_struct_name
+    )
+        *
```

Comment:
I think this is missing the additional condition that market_id is not null

Because a `MarketRegistrationEvent` that *de-lists* will have the base/quote fields non-null, but market ID, lot_size, tick_size, min_size, and underwriter_id will be null

The easiest way to verify that the event is a listing event instead of a delisting event is to enforce that market ID is non-null

Reply:
I see what you mean and it's partially true.

If I add `market_id IS NOT NULL`, and then I query this, I'll ignore the latest delisting event, and I'll return a market that has been delisted.

But leaving it as is will return rows which contain `market_id = NULL`, which might not be what we want.

I'll update